### PR TITLE
screen orientation: test non-fully active behavior

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

File: screen-orientation/non-fully-active.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<p id="#fragment"></p>
```
Comment (marcoscaceres) [inline screen-orientation/non-fully-active.html:6]:
```suggestion
```

File: screen-orientation/non-fully-active.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    );
+  }, "Attempting to lock non-fully active documents results in a InvalidStateError");
+
+  promise_test(async (t) => {
+    const iframe = await attachIFrame();
+    const { orientation } = iframe.contentWindow.screen;
+
+    iframe.remove();
+
+    await assert_throws_dom(t, "InvalidStateError", orientation.unlock());
```
Comment (cdumez) [inline screen-orientation/non-fully-active.html:39]: Is the await really useful here?

File: screen-orientation/non-fully-active.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    await test_driver.bless("request full screen", null, iframe.contentWindow);
+    await iframe.contentDocument.documentElement.requestFullscreen();
+
+    const p = orientation.lock(opposite);
+
+    iframe.remove();
+
+    await promise_rejects_dom(t, "AbortError", p);
+    await assert_throws_dom(t, "InvalidStateError", orientation.unlock());
```
Comment (cdumez) [inline screen-orientation/non-fully-active.html]: Is the await really useful here?

File: screen-orientation/non-fully-active.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (marcoscaceres) [inline screen-orientation/non-fully-active.html:58]:
```suggestion
     assert_throws_dom(t, "InvalidStateError", ()=>{ orientation.unlock() });
```

File: screen-orientation/non-fully-active.html [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<p id="#fragment"></p>
+<script>
+  async function attachIFrame() {
+    const iframe = document.createElement("iframe");
+    document.body.appendChild(iframe);
+    await new Promise((resolve) => {
+      iframe.onload = resolve;
+      iframe.src = "resources/empty.html";
```
Comment (cdumez) [inline screen-orientation/non-fully-active.html:12]: Any reason we can't use about:blank?

File: screen-orientation/non-fully-active.html [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+    await test_driver.bless("request full screen", null, iframe.contentWindow);
+    await iframe.contentDocument.documentElement.requestFullscreen();
+
+    const p = orientation.lock(opposite);
+
+    iframe.remove();
+
+    await promise_rejects_dom(t, "AbortError", p);
+     assert_throws_dom(t, "InvalidStateError", () => { orientation.unlock() });
```
Comment (marcoscaceres) [inline screen-orientation/non-fully-active.html:58]:
```suggestion
    assert_throws_dom(t, "InvalidStateError", () => { orientation.unlock() });
```

File: screen-orientation/non-fully-active.html [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    );
+  }, "Attempting to lock non-fully active documents results in a InvalidStateError");
+
+  promise_test(async (t) => {
+    const iframe = await attachIFrame();
+    const { orientation } = iframe.contentWindow.screen;
+
+    iframe.remove();
+
+    await assert_throws_dom(t, "InvalidStateError", orientation.unlock());
```
Comment (marcoscaceres) [inline screen-orientation/non-fully-active.html:39]:
```suggestion
    assert_throws_dom(t, "InvalidStateError", () => { orientation.unlock() });
```

File: screen-orientation/non-fully-active.html (hunk 5 above)
Comment (marcoscaceres) [reply screen-orientation/non-fully-active.html:12]: Was unsure about same origin check. Can use about:blank if it's treated as same origin.

**Review by cdumez — APPROVED** [review]
LGTM

**Pull request merged**